hilti/toolchain/include/base/intrusive-ptr.h

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2020 by the Zeek Project. See LICENSE for details.
```

Comment:
Why do we need to duplicate this type into this namespace?

Reply:
For the most part, the toolchain code doesn't use the `hilti::rt:*` namespace directly, but imports what it needs. Not a strong reason for doing so, just that it doesn't feel quite right to use "runtime" inside the toolchain. We are not fully consistent on that, and I wouldn't fight to hard to abandon the policy, but that's why I did it here as well (admittedly, this one one particular ugly because of the complex declarations.)